scripts/zones/Ifrits_Cauldron/mobs/Bomb_Queen.lua

```diff
@@ -1,35 +1,53 @@
 -----------------------------------
 -- Area: Ifrit's Cauldron
 --   NM: Bomb Queen
+--  Vid: https://www.youtube.com/watch?v=AVsEbYjSAHM
 -----------------------------------
 local ID = require("scripts/zones/Ifrits_Cauldron/IDs")
 require("scripts/globals/status")
 -----------------------------------
 
 function onMobInitialize(mob)
+    mob:setMobMod(dsp.mobMod.IDLE_DESPAWN, 180)
     mob:setMobMod(dsp.mobMod.HP_STANDBACK, -1)
     mob:setMobMod(dsp.mobMod.DRAW_IN, 1)
-    mob:setMobMod(dsp.mobMod.IDLE_DESPAWN, 180)
+    mob:setMod(dsp.mod.STUNRES, 50)
 end
 
 function onMobSpawn(mob)
-    mob:addMod(dsp.mod.STUNRES, 50)
+    mob:setLocalVar("petCooldown", os.time() + 5) -- five seconds for first pet
 end
 
-function onMobFight(mob,target)
-    local battleTime = mob:getBattleTime()
+function onMobFight(mob, target)
+    -- every ~20 seconds after first pet, a new pet will spawn around queen's position.
+    -- 50% chance for Prince/Princess. 1% chance for Bastard.
 
-    if (battleTime - 5) % 20 == 0 then -- every 20 seconds after 5 seconds, source: https://www.youtube.com/watch?v=AVsEbYjSAHM
-        local random = math.random(1,100);
-        local petId = mob:getID() + 1 + math.floor(random / 25); -- approximately 1% chance to spawn bomb bastard
+    local mobId = mob:getID()
+
+    if os.time() > mob:getLocalVar("petCooldown") then
+        local petId = mob:getID() + 1 + math.floor(math.random(0, 100) / 25)
         local pet = GetMobByID(petId)
-        local pos = mob:getPos()
-        pet:setSpawn(unpack(pos))
-        pet:spawn()
-        pet:updateEnmity(target)
-    end
 
+        if pet and not pet:isSpawned() then
```

Comment:
if pet is spawned there is a problem, since the pet should have blown up by now

Reply:
10 second wait until using ability + the time it takes to do the ability + the time it takes to despawn the mob, can be greater than 20 seconds.  Even in the videos, you'll see a pet spawned while the previous pet is still despawning.  Also, pets could be slept or otherwise prevented from using their move at exactly 10 seconds.